tests/javascript_tests/test_util.js

```diff
@@ -118,6 +118,47 @@ requirejs(['jquery', 'underscore', 'util'], function($, _, util) {
 
     });
 
+    /**
+     *
+     * Test that strings like "Infinity" are sorted with words in
+     * naturalSort's output, rather than with numbers. This test was adapted
+     * from Empress.
+     *
+     */
+    test("Test that naturalSort doesn't treat Infinity / NaN as numbers", function () {
+      var eles = [
+        "1",
+        "2",
```

Comment:
is it important to include things like 1e10?

Reply:
I hadn't thought of that! Would be nice to have tested. That being said, it looks like in JS scientific notation stuff like 1e10 is interpreted correctly as a number by `parseFloat()`, so I think this should be handled as expected already (but ofc would be best to explicitly test for that :)

```js
> parseFloat("1e10");
10000000000
```

Reply:
Excellent :)